### PR TITLE
Implement Minder TestKit

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

TestKit is an implementation of several interfaces that allows for
easier testing. The intent is to be able to test rule types locally via
a programmatic framework, as opposed to relying on an integration test.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/actions/remediate/pull_request/pull_request.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ import (
 	"github.com/mindersec/minder/internal/profiles/models"
 	"github.com/mindersec/minder/internal/util"
 	pb "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	interfaces2 "github.com/mindersec/minder/pkg/engine/v1/interfaces"
```
Comment (jhrozek) [inline internal/engine/actions/remediate/pull_request/pull_request.go:43]: this is irritating about goland, can you rename the import name? (You can use the refactor action on `interfaces2`)

File: internal/engine/ingester/builtin/builtin.go [hunk 2]
```diff
@@ -21,15 +21,15 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/mindersec/minder/pkg/engine/v1/interfaces"
+	"github.com/mindersec/minder/pkg/entities/v1/checkpoints"
```
Comment (jhrozek) [inline internal/engine/ingester/builtin/builtin.go:25]: does this order of imports lint?

**Review by jhrozek — DISMISSED** [review]
the interfaces are nice! Feel free to fix the misnamed imports in a follow-up

File: internal/engine/actions/remediate/pull_request/pull_request.go (hunk 1 above)
Comment (JAORMX) [reply internal/engine/actions/remediate/pull_request/pull_request.go:43]: Will do

**Review by jhrozek — DISMISSED** [review]

**Review by jhrozek — APPROVED** [review]

**Pull request merged**